sycl/doc/syclgraph/SYCLGraphUsageGuide.md

```diff
@@ -222,6 +222,90 @@ q.submit([&](sycl::handler &CGH) {
 });
 ```
 
+### Guidance For Library Authors
+
+In addition to the general SYCL-graph compatibility guidelines there are some
+considerations that are more relevant to library authors to be compatible with
+SYCL-Graph and allow seamless capturing of library calls in a graph.
+
+#### Graph-owned Memory Allocations For Temporary Memory
+
+A common pattern in libraries with specialized SYCL kernels can involve the
+allocation and use of temporary memory for those kernels. One approach is custom
+allocators which rely on SYCL events to control the lifetime and re-use of this
+temporary memory, but these are not compatible with events returned from queue
+submissions which are recorded to a graph. Instead the
+[sycl_ext_oneapi_async_memory_alloc](../extensions/experimental/sycl_ext_oneapi_async_memory_alloc.asciidoc)
+extension can be used which provides similar functionality for eager SYCL usage
+as well as compatibility with graphs.
+
+When captured in a graph calls to these extension functions create graph-owned
+memory allocations which are tied to the lifetime of the graph. These
+allocations can be created as needed for library kernels and the SYCL runtime
+may be able to re-use memory where appropriate to minimize the memory footprint
+of the graph. This can avoid the need for a library to manage the lifetime of
+these allocations themselves, or be aware of the library calls being recorded to
+a graph.
+
+It is important to set correct dependencies between allocation commands, kernels
```

Comment:
I'd maybe add here that the same is also achievable via correct ordering with in-order queues.

Reply:
Hmm, I feel like we could assume that a user knows how to create SYCL dependencies here. Maybe changing `set` to `ensure` would be better though? I feel like that better implies using in-order queues or explicitly setting dependencies yourself.

Reply:
Yeah I don't think we need to be explicit here about the different ways to create dependencies, just word it such that it covers both cases, I think changing `set` to `ensure` would better imply both options.